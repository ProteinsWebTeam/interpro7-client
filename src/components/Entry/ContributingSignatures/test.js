// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { ContributingSignatures } from '.';

const renderer = new ShallowRenderer();

describe('<ContributingSignatures />', () => {
  test('should render', () => {
    renderer.render(
      <ContributingSignatures
        contr={{
          pfam: { PF00051: 'Kringle domain' },
        }}
        data={{
          loading: false,
          payload: {
            databases: {
              pfam: {
                canonical: 'pfam',
                description:
                  'Pfam is a large collection of multiple sequence alignments and hidden Markov models covering many common protein domains. Pfam is based at EMBL-EBI, Hinxton, UK.',
                name: 'Pfam',
                releaseDate: '2018-10-02T00:00:00Z',
                type: 'entry',
                version: '32.0',
              },
            },
          },
        }}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
