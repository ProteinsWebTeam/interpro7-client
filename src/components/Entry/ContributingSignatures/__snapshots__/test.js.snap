// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContributingSignatures /> should render 1`] = `
<div
  className="side-panel margin-top-small margin-bottom-large"
>
  <div
    className="md-icon-list-box margin-bottom-large"
  >
    <h5>
      Contributing Member Database Entr
      y
    </h5>
    <ul
      className="md-list"
    >
      <li>
        <MemberSymbol
          className="md-small"
          svg={false}
          type="pfam"
        />
        <div>
          <span
            className="db-name"
          >
            Pfam
            :
             
          </span>
           
          <React.Fragment>
            <Memo(Loadable)
              data={
                Object {
                  "db": "pfam",
                  "name": "PF00051",
                }
              }
              processData={[Function]}
            />
            
            <Memo(SignatureLink)
              accession="PF00051"
              db="pfam"
              label="Kringle domain"
            />
          </React.Fragment>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
